torch/csrc/api/include/torch/nn/modules/sequential.h

```diff
@@ -175,18 +175,7 @@ class SequentialImpl : public Cloneable<SequentialImpl> {
   /// Adds a new (boxed) `Module` to the `Sequential` container.
   template <typename ModuleType>
   void push_back(std::shared_ptr<ModuleType> module_ptr) {
-    // Nesting Sequential doesn't work because `forward()`'s return type is
-    // templatized, so it'll give a nasty compiler error.
-    static_assert(
-        !std::is_same<SequentialImpl, ModuleType>::value,
-        "Sequential is not nestable");
-    static_assert(
-        torch::detail::is_module<ModuleType>::value,
-        "Can only add objects derived from nn::Module to Sequential");
-    static_assert(
-        torch::detail::has_forward<ModuleType>::value,
-        "Can only add modules with a forward() method to Sequential");
-    push_back(AnyModule(std::move(module_ptr)));
+    push_back(modules_ordered_dict({{std::to_string(modules_.size()), std::move(module_ptr)}}));
```

Comment:
I don't understand why sequential needs to know anything about modules_ordered_dict or named modules -- can you not just add new overloads for all existing `push_back`s, move all the checking to those functions, and just have the existing functions forward to the new ones, passing the right name?  And an OrderedDict constructor overload that extracts the modules and checks them in the standard way.

Reply:
Fixed